docs/DeveloperGuide.md

```diff
@@ -1,22 +1,222 @@
 # Developer Guide
 
 ## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+ChatParser structure inspired by: 
+[this repository](https://github.com/kennethSty/ip)
+Calorie consumption bar inspired by 
+[this blogpost](https://medium.com/javarevisited/how-to-display-progressbar-on-the-standard-console-using-java-18f01d52b30e)
 
 ## Design & implementation
+### High Level Class Design
+The main classes of this implementation are:
+- HealthMate
+- ChatParser
+- User
+- MealList
+- MealEntriesList
+- Meal
+- MealEntry
+- HistoryTracker
+- UI
+
+In each class we focused on maintaining a tight abstraction barrier between classes.
+This specifically includes adherance to the "Tell Don't Ask" principle which was enforced by
+making most attributes of all classes above private and avoiding getter methods if possible.
+The following diagram illustrates the resulting associations, methods and attributes.
+For the sake of clarity, UI class is omitted since it is used across most classes to systematize any form of
```

Comment:
would it be helpful to include a minimal/simple representation of the UI class in the high level class diagram to indicate dependency across all classes. this might improve clarity around UI interactions